cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -242,11 +261,14 @@ public void visitEnd() {
             return;
           }
           AuthEnforceAnnotationNodeProcessor nodeProcessor =
-            new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
-          Map<String, Integer> paramAnnotation = processParameterAnnotationNode(parameterAnnotationNode);
+                  new AuthEnforceAnnotationNodeProcessor(authEnforceAnnotationNode);
+          Map<String, Integer> paramPositions = processParameterAnnotationNode(parameterDetails);
 
           List<EntityPartDetail> entityPartDetails = new ArrayList<>();
           for (String name : nodeProcessor.getEntities()) {
+            // just need to check for empty string since it cannot be null as that will be a compilation error
+            Preconditions.checkArgument(!name.isEmpty(), "Found an empty string in entity parts of annotation %s on " +
+              "method %s in class %s", nodeProcessor, methodName, className);
             // Make sure that the entities specified in the AuthEnforce annotation are found in method parameters
             // or class fields. Its fine if they exist at both places in that case we will give preference to
             // method parameters unless its been specified with this. in that case its always looked in class field.
```

Comment:
No need the `entityPart` variable. Just add to `entityPartDetails` in place. Less non-final variable is better